### PR TITLE
B dont ask login for public git

### PR DESCRIPTION
**Pull request by carebare47**
## Proposed changes

Public repos shouldn't need ssh auth to fetch/pull from github. github api rate limits requests to 60 per hour, so a workaround for this is included

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: bin/run-ansible.sh [hunk 1]
```diff
@@ -162,73 +162,160 @@ for extra_var in $extra_vars; do
...
+are_all_pr_repos_public(){
+    REPO_IS_PRIVATE="true"
+    printf '\n%s\n' "Testing if repos specified in pr_branches are all public" >&2
+    PR_BRANCHES="$@"
+    for i in $PR_BRANCHES; do
+      echo "Testing URL: ${i}" >&2
+      # Convert github URL to shadow-robot/repo_name
+      user_slash_repo=$(echo $i | sed -r 's/.*github\.com\///g' | sed -r s'/\/tree.*//g' | sed -r 's/\/pull.*//g')
+      REPO_IS_PUBLIC=$(is_repo_public $user_slash_repo)
+      if [[ $REPO_IS_PUBLIC == "false" ]]; then
```
Comment (dg-shadow) [inline bin/run-ansible.sh:203]: text compare of boolean :grimacing:  bash ftw

**Review by dg-shadow — APPROVED** [review]

File: bin/run-ansible.sh [hunk 2]
```diff
@@ -156,73 +156,160 @@ for extra_var in $extra_vars; do
...
+check_github_next_steps(){
+    PUBLIC_REPO_STATUS=$1
+    NEXT_STEP=0
+    if [[ $PUBLIC_REPO_STATUS == "403" ]]; then
+        printf '%s' "WARNING: Rate limit exceeded for github api requests." >&2
+        printf '%s\n' " It is not currently possible to confirm whether all the URLs specified in PR_BRANCHES belong to public repos" >&2
+        printf '%s\n' "(Rate limits only last for 60 minutes, if you are unsure then please try again later)" >&2
+        if [[ $(confirm "Would you like to create a key and authenticate it? y/N") == "y" ]]; then
+            NEXT_STEP="generate_key"
+        elif [[ $(confirm "Would you like to continue without this check? y/N") == "y" ]] ; then
```
Comment (niko-holmes) [inline bin/run-ansible.sh:217]: Is there a reason `y/N` is written as `y/N` and not `y/n` as in other parts of the script (e.g. line `286`)?

**Review by niko-holmes — APPROVED** [review]

File: bin/run-ansible.sh (hunk 2 above)
Comment (carebare47) [reply bin/run-ansible.sh:217]: capital refers to default value, there are two different implementations of confirmation in this script, wasn't sure it was the best use of time to refactor and test the existing confirmation but if you think that's best then you'd probably be correct so I'd be happy to

Comment (niko-holmes) [reply bin/run-ansible.sh:217]: Thanks for clarifying, wasn't aware that was a convention. I agree that it's probably not worth the time commitment to refactor and test the existing confirmations.

**Pull request merged**